otherlibs/unix/read_win32.c

```diff
@@ -103,3 +103,34 @@ CAMLprim value caml_unix_read_bigarray(value fd, value vbuf,
   }
   CAMLreturn(Val_int(numread));
 }
+
+intnat caml_unix_fast_read(value fd, value buf, intnat ofs, intnat len)
+{
+  intnat ret = -1;
+
+  if (Descr_kind_val(fd) == KIND_SOCKET) {
+    SOCKET s = Socket_val(fd);
+    ret = recv(s, &Byte(buf, ofs), len, 0);
+    if (ret == SOCKET_ERROR) ret = -1;
```

Comment:
The error handling here is missing something: in case of error you need to call `caml_win32_maperr(WSAGetLastError())` in order to set `errno` correctly. See what is done in the usual `read` call for inspiration.